src/Tooltip/Tooltip.js

```diff
@@ -0,0 +1,158 @@
+    height: 32,
+    lineHeight: '32px',
+    opacity: 0,
+    padding: '0 16px',
+    transform: 'scale(0)',
+    transition: theme.transitions.create(['opacity', 'transform'], {
+      duration: theme.transitions.duration.shortest,
+    }),
+    [theme.breakpoints.up('sm')]: {
+      height: 22,
+      lineHeight: '22px',
+      padding: '0 8px',
+      fontSize: 10,
+    },
+  },
+  tooltipLeft: {
+    marginRight: 24,
+    [theme.breakpoints.up('sm')]: {
+      marginRight: 14,
+    },
+  },
+  tooltipRight: {
+    marginLeft: 24,
+    [theme.breakpoints.up('sm')]: {
+      marginLeft: 14,
+    },
+  },
+  tooltipTop: {
+    marginBottom: 24,
+    [theme.breakpoints.up('sm')]: {
+      marginBottom: 14,
+    },
+  },
+  tooltipBottom: {
+    marginTop: 24,
+    [theme.breakpoints.up('sm')]: {
+      marginTop: 14,
+    },
+  },
+  tooltipOpen: {
+    opacity: 0.9,
+    transform: 'scale(1)',
+  },
+});
+
+function Tooltip(props) {
+  const {
+    children,
+    classes,
+    label,
+    open,
+    placement,
+    onRequestClose,
+    onRequestOpen,
+    className: classNameProp,
+    ...other
+  } = props;
+
+  const className = classNames(classes.root, classNameProp);
+
+  return (
+    <Manager {...other} className={className}>
+      <Target>
+        <div
+          onMouseOver={onRequestOpen}
+          onFocus={onRequestOpen}
+          onMouseLeave={onRequestClose}
+          onBlur={onRequestClose}
+        >
+          {children}
+        </div>
+      </Target>
+      <Popper placement={placement} className={classes.popper}>
+        <div
+          className={classNames(
+            classes.tooltip,
+            { [classes.tooltipTop]: placement === 'top' },
+            { [classes.tooltipBottom]: placement === 'bottom' },
+            { [classes.tooltipLeft]: placement === 'left' },
+            { [classes.tooltipRight]: placement === 'right' },
+            { [classes.tooltipOpen]: open },
+          )}
+        >
+          {label}
+        </div>
+      </Popper>
+    </Manager>
+  );
+}
+
+Tooltip.propTypes = {
+  /**
+   * Tooltip reference component
+   */
+  children: PropTypes.node,
+  /**
+   * Useful to extend the style applied to components.
+   */
+  classes: PropTypes.object.isRequired,
+  /**
+   * @ignore
+   */
+  className: PropTypes.string,
+  /**
+   * If `true`, Tooltip label
+   */
+  label: PropTypes.string,
+  /**
+   * Callback fired when the tooltip requests to be closed.
+   *
+   * @param {object} event The event source of the callback
+   */
+  onRequestClose: PropTypes.func,
+  /**
+   * Callback fired when the tooltip requests to be closed.
+   *
+   * @param {object} event The event source of the callback
+   */
+  onRequestOpen: PropTypes.func,
+  /**
+   * If `true`, the tooltip is shown.
+   */
+  open: PropTypes.bool,
+  /**
+   * If `true`, Tooltip label
+   */
+  placement: PropTypes.oneOf(['left', 'right', 'top', 'bottom']),
+};
+
+Tooltip.defaultProps = {
+  label: '',
+  open: false,
```

Comment:
I have the feeling that we could support an uncontrolled behavior by default, but still allow it to be controlled.. I feel that it will best cover 80% of the use cases.

Reply:
I would move the existing demo to be uncontrolled and add one more demo controlled.

Reply:
You can find some example of components that can be controlled and uncontrolled in the codebase.

Reply:
Like the SwitchBase or the Input.